gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/QuantityMapper.java

```diff
@@ -0,0 +1,68 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Quantity;
+import org.hl7.v3.PQInc;
+import org.hl7.v3.PQR;
+import org.hl7.v3.Value;
+
+public class QuantityMapper {
+    private static final String UNIT_SYSTEM = "http://unitsofmeasure.org";
+    private static final String TYPE_IVL_PQ = "IVL_PQ";
+
+
+    public Quantity mapQuantity(Value value) {
+        Quantity quantity = new Quantity();
+
+        if (TYPE_IVL_PQ.equals(value.getType())) {
+            if (value.getHigh() != null) {
+                setHighComparator(quantity, value.getHigh());
+            } else if (value.getLow() != null) {
+                setLowComparator(quantity, value.getLow());
+            }
+        } else {
+            setQuantityValueAndUnit(quantity, value.getValue(), value.getUnit(), value.getTranslation());
+        }
+
+        return quantity;
+    }
+
+    private void setUnit(Quantity quantity, String unit, List<PQR> translation) {
+        if (StringUtils.isNotBlank(unit)) {
+            if (translation != null && !translation.isEmpty()) {
+                quantity.setUnit(translation.get(0).getOriginalText());
+            } else {
+                quantity.setUnit(unit);
+                quantity.setSystem(UNIT_SYSTEM);
+                quantity.setCode(unit);
+            }
+        }
+    }
+
+    private void setHighComparator(Quantity quantity, PQInc high) {
```

Comment:
Maybe `setQuantityWithHighComparator` would be more descriptive, as this method also sets other Quantity fields? And the same for `setLowComparator`